contracts/CampaignFactory.sol

```diff
@@ -22,6 +23,10 @@ contract CampaignFactory {
 }
 
 contract Campaign {
+    struct Contributer {
+        bool hasFundBefore;
+        bool isApprover;
```

Comment:
I think we also need to save total balance of all contributions done by this person